### PR TITLE
Reduce DB size

### PR DESCRIPTION
**Pull request by antooni**
STATS:
- it took ~30 mins to run locally

We only need to store:
- hourly data for last 7D
- sixHourly data for last 90D
- daily data for whole time range

This is why we can get rid of the earlier hourly and sixHourly data that is no longer being used by the chart. 

**How?**
- add to every tvl repository method to remove such historical data
- add TvlCleaner class that will function as a scheduler of deletion
- change the way the clock ticks

Resolve L2B-2296

15:17:32.798 INFO [ TvlCleaner ] Cleaning BlockNumberRepository
15:17:32.800 INFO [ AggregatedReportUpdater ] Started
15:17:32.800 INFO [ TvlModule ] Started
15:17:32.800 INFO [ StatusModule ] Started
15:17:32.821 INFO [ TvlCleaner ] Cleaned BlockNumberRepository
    { hourly: 0, sixHourly: 0 }
15:17:32.821 INFO [ TvlCleaner ] Cleaning PriceRepository
15:17:33.952 INFO [ PriceUpdater ] Update completed
    { timestamp: 1706706000 }
15:17:34.373 INFO [ TvlCleaner ] Cleaned PriceRepository
    { hourly: 0, sixHourly: 0 }
15:17:34.374 INFO [ TvlCleaner ] Cleaning BalanceRepository
15:36:43.724 INFO [ TvlCleaner ] Cleaned BalanceRepository
    { hourly: 168731500, sixHourly: 22354522 }
15:36:43.728 INFO [ TvlCleaner ] Cleaning TotalSupplyRepository
15:36:43.811 INFO [ TvlCleaner ] Cleaned TotalSupplyRepository
    { hourly: 74303, sixHourly: 8049 }
15:36:43.812 INFO [ TvlCleaner ] Cleaning CirculatingSupplyRepository
15:36:43.909 INFO [ TvlCleaner ] Cleaned CirculatingSupplyRepository
    { hourly: 76274, sixHourly: 9360 }
15:36:43.909 INFO [ TvlCleaner ] Cleaning ReportRepository
15:52:57.163 INFO [ ApiServer ]
    { type: 'request', method: 'GET', url: '/api/tvl/' }
15:53:01.854 INFO [ ApiServer ]
    {
      type: 'response',
      status: 200,
      timeMs: 4692,
      method: 'GET',
      url: '/api/tvl/'
    }
15:53:06.313 INFO [ TvlCleaner ] Cleaned ReportRepository
    { hourly: 287522756, sixHourly: 40654186 }
15:53:06.313 INFO [ TvlCleaner ] Cleaning AggregatedReportRepository
15:53:19.377 INFO [ TvlCleaner ] Cleaned AggregatedReportRepository
    { hourly: 15417360, sixHourly: 2187360 }

### CODE REVIEW

File: packages/backend/src/config/makeConfig.ts [hunk 1]
```diff
@@ -140,6 +140,7 @@ export function makeConfig(
     },
     chains: chains.map((x) => ({ name: x.name, chainId: ChainId(x.chainId) })),
     flags: flags.getResolved(),
+    tvlCleanerEnabled: flags.isEnabled('tvlCleaner'),
```
Comment (sz-piotr) [inline packages/backend/src/config/makeConfig.ts:143]: This has to be called before `flags.getResolved()`!

File: packages/backend/src/modules/tvl/TvlCleaner.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  deleteSixHourlyUntil: (timestamp: UnixTime) => Promise<number>
+}
+
+export class TvlCleaner {
+  private readonly taskQueue: TaskQueue<void>
+
+  constructor(
+    private readonly clock: Clock,
+    private readonly logger: Logger,
+    private readonly tables: Table[],
```
Comment (sz-piotr) [inline packages/backend/src/modules/tvl/TvlCleaner.ts:20]: tables -> repositories

File: packages/backend/src/peripherals/database/CirculatingSupplyRepository.test.ts [hunk 3]
```diff
@@ -35,6 +36,7 @@ describeDatabase(CirculatingSupplyRepository.name, (database) => {
   ]
 
   beforeEach(async () => {
+    // TODO: get rid of this, adding new tests is tricky because of it
     await repository.addMany(DATA)
```
Comment (sz-piotr) [inline packages/backend/src/peripherals/database/CirculatingSupplyRepository.test.ts:40]: Move this to each test that needs it

File: packages/backend/src/peripherals/database/BlockNumberRepository.test.ts [hunk 4]
```diff
@@ -12,6 +16,10 @@ describeDatabase(BlockNumberRepository.name, (database) => {
     await repository.deleteAll()
   })
 
+  afterEach(async () => {
```
Comment (sz-piotr) [inline packages/backend/src/peripherals/database/BlockNumberRepository.test.ts:19]: Do we need both beforeEach and afterEach? why?

File: packages/backend/src/peripherals/database/AggregatedReportRepository.ts [hunk 5]
```diff
@@ -181,10 +180,6 @@ export class AggregatedReportRepository extends BaseRepository {
   async addOrUpdateMany(reports: AggregatedReportRecord[]) {
     const rows = reports.map(toRow)
     const knex = await this.knex()
-    const timestampsMatch = reports.every((r) =>
```
Comment (sz-piotr) [inline packages/backend/src/peripherals/database/AggregatedReportRepository.ts:184]: Why this change?

File: packages/backend/src/modules/tvl/TvlCleaner.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    })
+  }
+
+  async clean() {
+    const hourlyDeletionBoundary =
+      this.clock._TVL_ONLY_getHourlyDeletionBoundary()
+    const sixHourlyDeletionBoundary =
+      this.clock._TVL_ONLY_getSixHourlyDeletionBoundary()
+
+    for (const table of this.tables) {
```
Comment (sz-piotr) [inline packages/backend/src/modules/tvl/TvlCleaner.ts:46]: Maybe we can use Promise.all here? Those are separate tables after all

File: packages/backend/src/modules/tvl/TvlCleaner.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      )
+      this.logger.info(`Cleaned ${table.constructor.name}`, {
+        hourly,
+        sixHourly,
+      })
+    }
+  }
+}
+
+export function initializeTvlCleaner(
```
Comment (sz-piotr) [inline packages/backend/src/modules/tvl/TvlCleaner.ts:60]: I think maybe we want to make it explicit in the module instead rather than depending on it here.

**Review by sz-piotr — CHANGES_REQUESTED** [review]

File: packages/backend/src/modules/tvl/TvlCleaner.ts (hunk 6 above)
Comment (antooni) [reply packages/backend/src/modules/tvl/TvlCleaner.ts:46]: I am a bit afraid of over-using our DB

File: packages/backend/src/peripherals/database/AggregatedReportRepository.ts (hunk 5 above)
Comment (antooni) [reply packages/backend/src/peripherals/database/AggregatedReportRepository.ts:184]: This is no longer needed and would make tests unnecessary call addOrUpdate for every entry seperately, I can change if you want to, but this assert is useless IMO

File: packages/backend/src/modules/tvl/TvlCleaner.ts (hunk 6 above)
Comment (antooni) [reply packages/backend/src/modules/tvl/TvlCleaner.ts:46]: This will run alongside our regular backend, I do not want it to take all the connections for some time.
